File: packages/rook_sdk_health_connect/lib/src/data/mapper/sync_instruction_mappers.dart
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk_health_connect/src/domain/enums/hc_sync_instruction.dart';

extension SyncStatusMappers on HCSyncInstruction {
  SyncInstructionProto toProto() {
    return switch (this) {
      HCSyncInstruction.syncOldest => SyncInstructionProto.SYNC_OLDEST,
      HCSyncInstruction.syncLatest => SyncInstructionProto.SYNC_LATEST,
      _ => SyncInstructionProto.NOTHING,
    };
  }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/SyncInstructionMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.SyncInstruction
import com.rookmotion.rook_sdk_health_connect.data.proto.SyncInstructionProto

fun SyncInstructionProto.toDomain(): SyncInstruction {
    return when (this) {
        SyncInstructionProto.NOTHING -> SyncInstruction.NOTHING
        SyncInstructionProto.SYNC_OLDEST -> SyncInstruction.SYNC_OLDEST
        else -> SyncInstruction.SYNC_LATEST
    }
}
